validation/immersed_boundaries/flow_over_hills.jl

```diff
@@ -0,0 +1,160 @@
+using Printf
+using Statistics
+using Oceananigans
+using Oceananigans.ImmersedBoundaries: ImmersedBoundaryGrid, GridFittedBottom, mask_immersed_field!
+
+@inline bottom_drag_func(x, y, t, u, Cᴰ) = - Cᴰ * u^2
+
+function hilly_simulation(;
+                          Nx = 64,
+                          Nz = Nx,
+                          ϵ = 0.1,
+                          Re = 1e4,
+                          N² = 1e-2,
+                          bottom_drag = false,
+                          stop_time = 1,
+                          save_interval = 0.1,
+                          architecture = CPU(),
+                          name = "flow_over_hills")
+
+    underlying_grid = RectilinearGrid(architecture,
+                                      size = (Nx, Nz),
+                                      halo = (3, 3),
+                                      x = (0, 2π),
+                                      z = (0, 2π),
+                                      topology = (Periodic, Flat, Bounded))
```

Comment:
I think in a 2D domain the effect of form drag will probably be exacerbated in comparison with a 3D simulation. I think this is find since this is just a test, but something to keep in mind

Reply:
Ok!